chunky/src/java/se/llbit/math/BVH.java

```diff
@@ -341,6 +360,7 @@ private Node constructSAH_MA(Primitive[] primitives) {
         Primitive[] chunk = chunks.pop();
         if (chunk.length < SPLIT_LIMIT) {
           nodes.push(new Leaf(chunk));
+          task.updateInterval(progress += chunk.length, startTime, 16384);
```

Comment:
```suggestion
          progress += chunk.length;
          task.updateInterval(progress, startTime, 16384);
```

Also, where does the magic `16384` number come from? Could it be moved to a constant with a name that explains it?

Reply:
`16384` was just a random number I chose that was low enough that it would still show progress, but not so high that it would hang / crash due to the amount of calls. The value probably depends on the application so I'm not sure moving it to a global constant is a good idea. I can add a comment explaining this.

Reply:
I changed the BVH progress to include loading primitives (which takes about half the time). To implement this, the task now goes from 0 to 1000 with progress updating every subdivision (0-500 on loading primitives, 500-1000 on building BVH).